installer/install.go

```diff
@@ -0,0 +1,397 @@
+	Stream      string       `json:"stream"`
+	Error       string       `json:"error"`
+	ErrorDetail *ErrorDetail `json:"errorDetail"`
+}
+
+type Options struct {
+	ImageRepo         string `short:"r" long:"image-repo" required:"false" description:"Private Docker image repo URL" value-name:"URL"`
+	ImageTar          string `short:"p" long:"image-tar" required:"false" description:"Path to image tar" value-name:"PATH"`
+	//ImageRepoUser     string `long:"repo-user" required:"false" description:"Private Docker image repo URL" value-name:"USER"`
+	//ImageRepoPw       string `long:"repo-pw" required:"false" description:"Private Docker image repo URL" value-name:"PASSWORD"`
+	ClusterName       string `short:"c" long:"cluster-name" required:"true" description:"Private cluster name" value-name:"NAME"`
+	Token     		  string `short:"t" long:"token" required:"true" description:"Astra API token" value-name:"TOKEN"`
+	AcceptEula        bool   `long:"accept-eula" required:"true" description:"(flag) Accept End User License Agreement"`
+	AstraAccountId    string `short:"a" long:"account-id" required:"true" description:"Astra account ID" value-name:"ID"`
+	AstraUrl          string `short:"u" long:"astra-url" required:"false" default:"https://eap.astra.netapp.io" description:"Url to Astra. E.g. 'https://integration.astra.netapp.io'" value-name:"URL"`
+	SkipTlsValidation bool   `short:"z" long:"disable-tls" required:"false" description:"(flag) Disable TLS validation. TESTING ONLY."`
+	HostAliasIP string `long:"hostAliasIP" required:"false" description:"The IP of the Astra host. TESTING ONLY." value-name:"IP"`
+	HostAlias bool   `long:"hostAlias" required:"false" description:"(flag) Set to enable HostAliasIP. TESTING ONLY"`
+
+}
+
+const (
+	NatsImageName               = "nats"
+	ConnectorDefaultsConfigPath = "./astraconnector_defaults.yaml"
+	YamlOutputPath              = "./deployConfig.yaml"
+	OperatorYamlPath            = "./astraconnector_operator.yaml"
+	ConnectorNamespace 			= "astra-connector"
+	ConnectorOperatorNamespace  = "astra-connector-operator"
+)
+
+func main() {
+	formatter := &log.TextFormatter{
+		FullTimestamp: true,
+	}
+	log.SetFormatter(formatter)
+
+	var opts Options
+	var connectorConfig ConnectorConfig
+	_, err := flags.Parse(&opts)
+	checkFatalErr(err)
+
+	log.Info("Installing Astra Connector")
+
+	absPath, err := filepath.Abs(ConnectorDefaultsConfigPath)
+	checkFatalErr(err)
+	yamlFile, err := ioutil.ReadFile(absPath)
+	checkFatalErr(err)
+	err = yaml.Unmarshal(yamlFile, &connectorConfig)
+	checkFatalErr(err)
+
+	// input check
+	if opts.ImageTar != "" || opts.ImageRepo != "" {
+		if !(opts.ImageTar != "" && opts.ImageRepo != "") {
+			log.Fatal("error: 'image-tar' and 'image-repo' options must be used together")
+		}
+	}
+
+	// input check
+	if opts.HostAlias {
+		if opts.HostAliasIP == "" {
+			log.Fatal("error: 'hostAliasIP' must be defined if the 'hostAlias' flag is set ")
+		}
+	}
+
+	if opts.ImageTar != "" && opts.ImageRepo != "" {
```

Comment:
Is this already tested for?
Also, for these sort of tests, best to trim off spaces.  And maybe check len

Reply:
Ah yeah, you're right. I just added a check for this above. I'll clean that up

I don't think trim is needed in this case. These args are coming from the go-flags package which will trim the whitespace unless the user escapes it or quotes it (how CLIs do)>